signbank/dictionary/tests/test_csv_import.py

```diff
@@ -143,8 +144,81 @@ def test_share_ids_existing_on_glosses_are_skipped(self):
         file = SimpleUploadedFile(
             content=data.read(), name=data.name, content_type="content/multipart"
         )
-        Gloss.objects.create(dataset=self.dataset, idgloss="Share:11", nzsl_share_id="12345")
+        gloss = Gloss.objects.create(dataset=self.dataset, idgloss="Share:11", nzsl_share_id="12345")
+        GlossVideo.objects.create(
+            gloss=gloss,
+            is_public=True,
+            dataset=Dataset.objects.create(name="testdataset2", signlanguage=self.signlanguage),
+            videofile=SimpleUploadedFile("testvid.mp4", b'data \x00\x01', content_type="video/mp4"),
+            video_type=FieldChoice.objects.first(),
+            title="Main"
+        )
+
+        response = self.client.post(
+            reverse('dictionary:import_nzsl_share_gloss_csv'),
+            {"dataset": self.dataset.pk, "file": file},
+            format="multipart"
+        )
+        self.assertEqual(response.status_code, 200)
+        session = self.client.session
+        self.assertEqual(self.dataset.pk, session["dataset_id"])
+        self.assertListEqual([csv_content[0]], session["glosses_new"])
+        self.assertListEqual([csv_content[1]], response.context["skipped_existing_glosses"])
+
+    def test_share_ids_existing_on_glosses_with_no_videos_are_not_skipped(self):
```

Comment:
suggestion: I think something like this would be a better name

```suggestion
    def test_share_ids_existing_on_glosses_with_no_videos_have_their_videos_reimported(self):
```

as that is the important part of this test

Reply:
Done!